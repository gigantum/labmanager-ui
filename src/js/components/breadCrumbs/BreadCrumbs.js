import React, { Component } from 'react'

export default class BreadCrumbs extends Component {
  constructor(props){
  	super(props);

    let paths = props.location.location.pathname.split('/')
    paths.shift()
    this.state = {
      breadCrumbs: paths
    }
  }

  componentWillReceiveProps(nextProps) {

    let paths = nextProps.location.location.pathname.split('/')
    paths.shift()
    this.setState({
      breadCrumbs: paths
    })
  }

  _reroute(bc){
      let index = this.state.breadCrumbs.indexOf(bc)
      let subArray = this.state.breadCrumbs.slice(0, index + 1)

      this.props.history.replace('/' + subArray.join('/')) //route to
  }

  _breadCrumb(bc){

    return (<li className="BreadCrumbs__list-item" key={bc} onClick={()=> this._reroute(bc)}>{bc}</li>)
  }

  render(){
    console.log(this.state)
    return(
        <div className="BreadCrumbs">
          <ul className="BreadCrumbs__list flex flex--row">
            {this.state.breadCrumbs.map((bc) => {
              return(this._breadCrumb(bc))
            })}
          </ul>
        </div>
      )
  }
}
