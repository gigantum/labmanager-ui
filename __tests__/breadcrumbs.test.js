import BreadCrumbs from './../src/js/components/breadCrumbs/breadCrumbs';
import React from 'react';
import renderer from 'react-test-renderer';
import history from './../src/js/history'

test('Test BreadCrumbs Rendering', () => {
      const component = renderer.create(<BreadCrumbs location={history} history={history} />
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
})

test('Test BreadCrumbs Rendering with pathname', () => {

      history.location.pathName = '/labbook/labook4'
      const component = renderer.create(<BreadCrumbs location={history} history={history} />
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
})
