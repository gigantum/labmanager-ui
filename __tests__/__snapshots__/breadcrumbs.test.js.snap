// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test BreadCrumbs Rendering 1`] = `
<div
  className="BreadCrumbs"
>
  <ul
    className="BreadCrumbs__list flex flex--row"
  >
    <li
      className="BreadCrumbs__list-item"
      onClick={[Function]}
    >
      
    </li>
  </ul>
</div>
`;

exports[`Test BreadCrumbs Rendering with pathname 1`] = `
<div
  className="BreadCrumbs"
>
  <ul
    className="BreadCrumbs__list flex flex--row"
  >
    <li
      className="BreadCrumbs__list-item"
      onClick={[Function]}
    >
      
    </li>
  </ul>
</div>
`;
